aten/src/ATen/native/TensorShape.cpp

```diff
@@ -840,6 +840,22 @@ Tensor repeat(const Tensor& self, IntArrayRef repeats) {
   return result;
 }
 
+Tensor tile(const Tensor& self, IntArrayRef reps){
+  // If self.size() > len(reps), reps is promoted to self.size() by pre-pending
+  // 1’s to it to keep the same behaviour as `numpy.tile`.
+  // Thus for a tensor of shape (2, 3, 4, 5), a dims of (2, 2) is treated
+  // as (1, 1, 2, 2).
+  int64_t size_diff = self.dim() - (int64_t)reps.size();
```

Comment:
const auto here would save this line from casting on the rhs, I think (if there is to be a cast it should be a C++-style cast, not a C-style cast)

Reply:
Right, const and `static_cast`! 
I tend to use `auto` only when a type is cumbersome (containers within containers for example), otherwise, I stick to typing myself so it's not hidden from future devs. But I know some schools prefer using `auto` as much as possible. Is it the case for PyTorch?

Reply:
We allow a lot of latitude for individual style. As long as your types are correct you can use whatever you like.